yarn-project/circuit-types/src/body.ts

```diff
@@ -77,9 +88,30 @@ export class Body {
     return new UnencryptedL2BlockL2Logs(logs);
   }
 
-  get numberOfTxs() {
-    // We gather all the txEffects that are not empty (the ones that have been padded by checking the first newNullifier of the txEffect);
-    return this.txEffects.reduce((acc, txEffect) => (!txEffect.nullifiers[0].equals(Fr.ZERO) ? acc + 1 : acc), 0);
+  /**
+   * Computes the number of transactions in the block including padding transactions.
+   * @dev Modified code from TxsDecoder.computeNumTxEffectsToPad
```

Comment:
We can't use the code from the specified fn because that is a Solidity function. I was not aware of the `ceilPowerOfTwo` so it didn't come to my mind to use it. It's a good idea to do it but I checked and the function is actually not used anywhere so I assume it is simply forgotten.

Since the function is not used and I think it's a good idea to use the same code logic as in Solidity I will nuke `ceilPowerOfTwo` function.

Good catch.

https://github.com/AztecProtocol/aztec-packages/pull/5482/commits/050fe849b918474d09c086cb0f50f9f4458e96e6